src/main/java/chess/dao/CommandDAO.java

```diff
@@ -0,0 +1,84 @@
+package chess.dao;
+
+import chess.dto.PointsDTO;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class CommandDAO {
+    private final String server;
+    private final String database;
+    private final String option;
+    private final String userName;
+    private final String password;
+
+    public CommandDAO(final String server, final String database, final String option, final String userName, final String password) {
+        this.server = server;
+        this.database = database;
+        this.option = option;
+        this.userName = userName;
+        this.password = password;
+    }
+
+    public Connection getConnection() {
+        Connection connection = null;
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            System.err.println(" !! JDBC Driver load 오류: " + e.getMessage());
+            e.printStackTrace();
+        }
+
+        try {
+            connection = DriverManager.getConnection("jdbc:mysql://" + server + "/" + database + option, userName, password);
+            System.out.println("정상적으로 연결되었습니다.");
+        } catch (SQLException e) {
+            System.err.println("연결 오류:" + e.getMessage());
+            e.printStackTrace();
+        }
+        return connection;
+    }
+
+    public void addCommand(final String roomId, final String startPoint, final String endPoint) {
+        String query = "INSERT INTO command(room_id, start_point, end_point) VALUES(?, ?, ?)";
+        try (PreparedStatement preparedStatement = getConnection().prepareStatement(query)) {
```

Comment:
지금은 Connection이 아닌 PreparedStatement만 close 하고 있네요!
가장 중요한 자원인 Connection도 close 해보면 어떨까요!?

Reply:
제가 [공식문서](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)를 제대로 읽지 않은것이 여기서 드러나네요

> In this example, the resource declared in the try-with-resources statement is a BufferedReader. The declaration statement appears within parentheses immediately after the try keyword. The class BufferedReader, in Java SE 7 and later, implements the interface java.lang.AutoCloseable. Because the BufferedReader instance is declared in a try-with-resource statement, it will be closed regardless of whether the try statement completes normally or abruptly (as a result of the method BufferedReader.readLine throwing an IOException).

try 안에서 **declared** 된것만 close를 해주는데, 저는 try안에서 connection을 리턴받으면 그것도 닫히는걸로 잘못 이해하고 있었네요 ㅎㅎ